StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/SpacingRules/SA1023CSharp9UnitTests.cs

```diff
@@ -3,9 +3,166 @@
 
 namespace StyleCop.Analyzers.Test.CSharp9.SpacingRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Testing;
     using StyleCop.Analyzers.Test.CSharp8.SpacingRules;
+    using Xunit;
+    using static StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly,
+        StyleCop.Analyzers.SpacingRules.TokenSpacingCodeFixProvider>;
 
     public class SA1023CSharp9UnitTests : SA1023CSharp8UnitTests
     {
+        [Fact]
+        public async Task TestFunctionPointerParameterValidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            await VerifyCSharpDiagnosticAsync(testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidPrecedingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int {|#0:*|}> FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotPreceded).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidTrailingSpaceAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int{|#0:*|} > FuncPtr;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr;
+}
+";
+
+            var expected = Diagnostic(DescriptorNotFollowed).WithLocation(0);
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeValidSpacingAsync()
```

Comment:
💡 This is actually a duplicate case from `TestFunctionPointerParameterInvalidTrailingSpaceAsync`, since the `fixedCode` is run through the analyzer automatically. If the test cases are grouped as described in the previous comment, the duplication will go away at the same time.